extensions/PhabBugz/lib/WebService.pm

```diff
@@ -98,15 +101,51 @@ sub revision {
     };
 }
 
+sub check_user_permission_for_bug {
+    my ($self, $params) = @_;
+
+    my $user = Bugzilla->login(LOGIN_REQUIRED);
+
+    # Ensure PhabBugz is on
+    ThrowUserError('phabricator_not_enabled')
+        unless Bugzilla->params->{phabricator_enabled};
+
+    # Validate that the requesting user's email matches phab-bot
+    ThrowUserError('phabricator_unauthorized_user')
+        unless $user->login eq PHAB_BMO_USER_EMAIL;
+
+    # Validate that a bug id and user id are provided
+    ThrowUserError('phabricator_invalid_request_params')
+        unless ($params->{bug_id} && $params->{user_id});
+
+    # Validate that the user and bug exist
+    Bugzilla::User->check({ id => $params->{user_id}, cache => 1 });
+
+    # Send back an object which says { "result": 1|0 }
+    return {
+        result => $user->can_see_bug($params->{bug_id})
```

Comment:
```
my $olduser = Bugzilla->user;
Bugzilla->set_user(Bugzilla::User->check({name => $email}));
...
Bugzilla->set_user($olduser);
```
But actually I need to provide a routine for this in the core API as you'll also need to clear a bunch of caches in some cases...